apps/lead/serializers.py

```diff
@@ -216,6 +216,10 @@ def validate(self, data):
             data, self.instance,
             File.objects.filter(pk=attachment_id).first()
         )
+        if self.instance and self.instance.is_assessment_lead:
+            raise serializers.ValidationError(
+                {'is_assessment_lead': 'Lead Assessment once set can\'t be changed. Please contact Admin'}
+            )
```

Comment:
We need to check if the lead already has an assessment before making it false. So,
- Always allow setting True.
- For setting False make sure there is no assessment attached to that lead.